### PR TITLE
Add warning before dumpwallet is executed

### PR DESCRIPTION
**Pull request by riordant**
PR for issue described here: https://github.com/zcoinofficial/zcoin/issues/290
Currently the text is @reubenyap 's warning.

### CODE REVIEW

**Comment by reubenyap** [comment]
@a-bezrukov Please double check if changing the dumpwallet command by adding a string at the end as per Yura's request or to add a dialog box would break any compatibility with Electrum / Insight before we merge this.

File: src/wallet/authhelper.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    code_type generateAuthorizationCode(code_type const & functionName)
+    {
+        rnd_type key = rnd_dist(generator);
+        key_store[functionName] = key;
+        return toCode(key);
+    }
+
+private:
+    std::random_device seed;
+    std::mt19937 generator;
```
Comment (psolstice) [inline src/wallet/authhelper.cpp]: There is (was?) a bug in mingw implementation of random_device
https://sourceforge.net/p/mingw-w64/bugs/338/
I need to double check if on Windows it returns always the same string

Comment (a-bezrukov) [reply src/wallet/authhelper.cpp]: Wow! Checking also.

Comment (psolstice) [reply src/wallet/authhelper.cpp]: Yes, it's always the same
There are two options
1. Use boost::random_device. This requires adding new dependency (libboost-random-dev)
2. Switch to hashing wallet data (keys)

Comment (a-bezrukov) [reply src/wallet/authhelper.cpp]: Changed it to be according to the 1st option.
LOCK2(cs_main, pwalletMain->cs_wallet);
generator.seed(pwalletMain->GetHDChain().masterKeyID.GetUint64(0) ^ time(NULL));

Comment (psolstice) [reply src/wallet/authhelper.cpp]: LGTM but please manually test if it gives different results on different Windows installations (even if the `time(NULL)` is exactly the same) before merging to master

**Comment by psolstice** [comment]
don't forget to manually test on Windows

**Comment by a-bezrukov** [comment]
Tested on Win7 and Win8, all cases provide random OTAC.

**Pull request merged**